os/arch/arm/src/s5j/s5j_pmstop.c

```diff
@@ -71,8 +71,20 @@
  * Description:
  *   Enter STOP mode.
  *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value.
+ *   In case of failure a negated errno value is returned to indicate the
+ *   cause of the failure.
+ *
  ****************************************************************************/
-void s5j_pmstop(void)
+int s5j_pmstop(void)
 {
-	s5j_clk_pll_select_mux(false);
+	/* FIXME: implement me */
+	asm("wfi");
```

Comment:
I am sorry but I can't understand relationship of up_idle and s5j_pmstop from your reply.
```
thapav 10 days ago  Member
This is provided by the SOC, but up_cpuidle() is called in case power framework is not supported.
```
1. In os_start.c, regardless of pm config, up_idle is called always.
2. When does s5j_pmstop need (or call)? when does up_idle?

Reply:
s5j_pmstop() is called when generic framework decides to switch to STOP PM state (one of the states provided by SoC) whereas up_idle will be called when there is no PM framework & no PM states (architecture specific WFI mode).

Will change current implementation to:
#ifdef CONFIG_PM
pm_idle();
#else
up_idle();
#endif